src/czar/HttpCzarWorkerModule.cc

```diff
@@ -0,0 +1,89 @@
+/*
+ * LSST Data Management System
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+// Class header
+#include "czar/HttpCzarWorkerModule.h"
+
+// System headers
+#include <set>
+#include <stdexcept>
+
+// Qserv headers
+#include "cconfig/CzarConfig.h"
+#include "global/intTypes.h"
+#include "http/Exceptions.h"
+#include "http/RequestQuery.h"
+#include "util/String.h"
+
+// LSST headers
+#include "lsst/log/Log.h"
+
+using namespace std;
+using json = nlohmann::json;
+
+namespace {
+LOG_LOGGER _log = LOG_GET("lsst.qserv.czar.HttpCzarWorkerModule");
+}
+
+namespace lsst::qserv::czar {
+
+void HttpCzarWorkerModule::process(string const& context, shared_ptr<qhttp::Request> const& req,
+                                   shared_ptr<qhttp::Response> const& resp, string const& subModuleName,
+                                   http::AuthType const authType) {
+    HttpCzarWorkerModule module(context, req, resp);
+    module.execute(subModuleName, authType);
+}
+
+HttpCzarWorkerModule::HttpCzarWorkerModule(string const& context, shared_ptr<qhttp::Request> const& req,
+                                           shared_ptr<qhttp::Response> const& resp)
+        : HttpModule(context, req, resp) {}
+
+json HttpCzarWorkerModule::executeImpl(string const& subModuleName) {
+    string const func = string(__func__) + "[sub-module='" + subModuleName + "']";
+    debug(func);
+    //&&&uj this seems irrelevant for a worker enforceInstanceId(func,
+    //cconfig::CzarConfig::instance()->replicationInstanceId());
+    enforceCzarName(func);
```

Comment:
The "instanceId" mechanism serves as a safeguard preventing "crosstalks" between two (or many) Qserv instances in case of misconfiguration when they run on the same network. That's why a value of this identifier is passed in all HTTP-based communications between services within Qserv (including Czar, workers, and various services of the Replication/Ingest system).

The same rule applies to the REST API version. Each time there are changes in expectations of a REST service (parameters added, or removed, or roles of parameters redefined) the API version must change and so must change the number mentioned in a code processing the corresponding requests:
```
checkApiVersion(__func__, 34);
```

Reply:
I missed this when it was posted. I'll need to add this.

Reply:
Fixed, but not pushed yet.